app/server/erlang/pi_server_cue.erl

```diff
@@ -29,99 +34,137 @@
         [log/1, log/2, debug/2, debug/3, debug/4]).
 
 
-start(InPort, CueHost, CuePort, Internal, Enabled) ->
-    Parent = self(),
-    Pid = spawn(fun() -> init(Parent, InPort, CueHost,
-                              CuePort, Internal, Enabled)
-                end),
-    register(?SERVER, Pid),
-    receive
-        {Pid, started} ->
-            Pid
-    end.
-
+server_name() ->
+    ?SERVER.
+
+start_link() ->
+    %% synchronous start of the child process
+    proc_lib:start_link(?MODULE, init, [self()]).
+
+
+init(Parent) ->
+    register(?SERVER, self()),
+    InPort = application:get_env(?APPLICATION, in_port, undefined),
+    CueHost = application:get_env(?APPLICATION, cue_host, {127,0,0,1}),
+    CuePort = application:get_env(?APPLICATION, cue_port, undefined),
+    Internal = application:get_env(?APPLICATION, internal, true),
+    Enabled = application:get_env(?APPLICATION, enabled, true),
+    io:format("~n"
+              "+--------------------------------------+~n"
+              "    This is the Sonic Pi OSC Server     ~n"
+              "       Powered by Erlang ~s             ~n"
+              "                                        ~n"
+              "        Incoming OSC on port ~p         ~n"
+              "  OSC cue forwarding to ~p              ~n"
+              "                     on port ~p         ~n"
+              "+--------------------------------------+~n~n~n",
+              [erlang:system_info(otp_release), InPort, CueHost, CuePort]),
 
-init(Parent, InPort, CueHost, CuePort, Internal, Enabled) ->
     case Internal of
         true ->
             {ok, InSocket} = gen_udp:open(InPort, [binary, {ip, loopback}]);
         _ ->
             {ok, InSocket} = gen_udp:open(InPort, [binary])
     end,
-    Parent ! {self(), started},
+
+    %% tell parent we have allocated resources and are up and running
+    proc_lib:init_ack(Parent, {ok, self()}),
+
     debug(2, "listening for OSC cues on socket: ~p~n",
           [try erlang:port_info(InSocket) catch _:_ -> undefined end]),
-    loop(InSocket, InPort, CueHost, CuePort, Internal, Enabled).
-
-loop(InSocket, InPort, CueHost, CuePort, Internal, Enabled) ->
+    State = #{parent => Parent,
+              enabled => Enabled,
+              cue_host => CueHost,
+              cue_port => CuePort,
+              internal => Internal,
+              in_port => InPort,
+              in_socket => InSocket
+             },
+    loop(State).
+
+loop(State) ->
     receive
-
         {udp, InSocket, Ip, Port, Bin} ->
             debug(3, "cue server got UDP on ~p:~p~n", [Ip, Port]),
             case (catch osc:decode(Bin)) of
```

Comment:
Shall one catch when suprvising. Especially when the clause for EXIT is missing :smile:

Reply:
Yes, this one is already on my list. :-)